larpandoracontent/LArVertex/TrainedVertexSelectionAlgorithm.cc

```diff
@@ -296,8 +296,8 @@ void TrainedVertexSelectionAlgorithm::GetEventShapeFeatures(const ClusterList &c
     }
 
     const float xSpan(this->GetCoordinateSpan(xMax, xMin));
-    const float ySpan(this->GetCoordinateSpan(yMax, zMin));
-    const float zSpan(this->GetCoordinateSpan(yMax, zMin));
+    const float ySpan(this->GetCoordinateSpan(yMax, yMin));
+    const float zSpan(this->GetCoordinateSpan(zMax, zMin));
 
     // Calculate the volume and longitudinality of the event (ySpan often 0 - to be investigated).
```

Comment:
There seems to be some 3D approach here applied to (currently always) a 2D input. In our usage, we would expect a `ySpan` (if implemented correctly, as modified above) to always be zero. That's because the 2D hits are always placed in a `y = constant = 0` plane. With `ySpan` always explicitly zero, this impacts the checks below.